### DISCUSSION TITLE
ZFS block allocation align to recordsize

### DISCUSSION THREAD

**Discussion by crossmeta** [openzfs/zfs#14883] [General]
Hello,
Is there any parameter to control zfs block offset to be aligned with the record size. For example if the record size is 1MB can the metaslab allocation align offset to 1MB . I think it is aligning to ashift now

Thanks

**Comment by amotin** [comment]
ZFS indeed aligns allocations only to ashift now.  There are some cases when physical ashift reported by device is too big to be used as logical ashift, so may be something still could be optimized there.  But your request makes no sense to me.
